src/test/java/com/cedarsoftware/util/io/WriteOptionsTest.java

```diff
@@ -1,289 +1,532 @@
+                Arguments.of(ZoneId.class),
+                Arguments.of(ZoneOffset.class)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("nonReferenceableClasses")
+    void testIsNonReferenceable_defaults(Class<?> c) {
+        WriteOptions options = new WriteOptionsBuilder().build();
+        assertTrue(options.isNonReferenceableClass(c)); // Assuming Date is a logical primitive
+    }
+
     @Test
-    void testIsNonReferenceableType() {
-        assertTrue(options.isNonReferenceableClass(Date.class)); // Assuming Date is a logical primitive
-        assertFalse(options.isNonReferenceableClass(Object.class)); // Assuming Object is not
+    void testIsNonReferenceable_notInDefaults() {
+        WriteOptions options = new WriteOptionsBuilder().build();
+        assertFalse(options.isNonReferenceableClass(NoTypeTest.Person.class)); // Assuming Date is a logical primitive
     }
 
-    // Test for getNonReferenceable method
+
     @Test
     void testGetNotReferenceableTypes() {
-        Collection<Class<?>> nonReferenceableClasses = options.getNonReferenceableClasses();
+        Collection<Class<?>> nonReferenceableClasses = new WriteOptionsBuilder().build().getNonReferenceableClasses();
         assertNotNull(nonReferenceableClasses);
-        assertTrue(nonReferenceableClasses instanceof LinkedHashSet);
     }
 
-    // Test for addNonReferenceable method
+
     @Test
-    void testAddNonReferenceableClass() {
-        options.addNonReferenceableClass(String.class);
-        assertTrue(options.isNonReferenceableClass(String.class));
+    void testAddNonReferenceableClass_isFound() {
+        WriteOptions options = new WriteOptionsBuilder().addNonReferenceableClass(MapOfMapsTest.Person.class).build();
+        assertTrue(options.isNonReferenceableClass(MapOfMapsTest.Person.class));
+    }
+
+    private static Stream<Arguments> aliasDefaults() {
+        return Stream.of(
+                Arguments.of("java.lang.Class", "class"),
+                Arguments.of("java.lang.String", "string"),
+                Arguments.of("java.util.Date", "date"),
+                Arguments.of("java.lang.Byte", "byte"),
+                Arguments.of("java.lang.Short", "short"),
+                Arguments.of("java.lang.Integer", "int"),
+                Arguments.of("java.lang.Long", "long"),
+                Arguments.of("java.lang.Float", "float"),
+                Arguments.of("java.lang.Double", "double"),
+                Arguments.of("java.lang.Character", "char"),
+                Arguments.of("java.lang.Boolean", "boolean")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("aliasDefaults")
+    void testAliasTypeNames_defaults(String fqName, String shortName) {
+        WriteOptions options = new WriteOptionsBuilder().build();
+        assertThat(options.getTypeNameAlias(fqName)).isEqualTo(shortName);
+    }
+
+    private static Stream<Arguments> aliasWithExtended() {
+        return Stream.of(
```

Comment:
Not sure if you want to be fully complete here (and match the extendedAliases.txt here) or just test a high percentage of them.  The extendedAliases file is the one above, that had about an additional 30 aliases that did not make it into this PR - so if when you add those to the .txt file, do you want them here?  Don't know, just asking.